flow/connectors/utils/partition.go

```diff
@@ -356,8 +341,6 @@ func (p *PartitionHelper) getPartitionForStartAndEnd(start any, end any) (*proto
 		return createTimePartition(v, end.(time.Time)), nil
 	case pgtype.TID:
 		return createTIDPartition(v, end.(pgtype.TID)), nil
-	case bson.ObjectID:
```

Comment:
do we not want to use partition helper for Mongo for consistency across sources? also thinking if we ever want to partition on other fields other than _id (i.e. with flattened mode in the future),  but not a big deal, we can add later too.

Reply:
`partitionHelper` can be used; it's just a bit cludgy to use it here (`bson.ObjectID` is not the type stored in protobufs and the helper tends to use types to figure out the type of range, which is `string`)

It doesn't give us any advantage here since we don't coalesce partitions together based on ObjectID range (could be done down the line)